frog/responsive-images.rkt

```diff
@@ -0,0 +1,116 @@
+#lang racket/base
+
+(require net/url
+         racket/contract/base
+         racket/contract/region
+         racket/file
+         racket/function
+         racket/list
+         racket/path
+         racket/port
+         racket/system
+         racket/string
+         "params.rkt"
+         "util.rkt"
+         "verbosity.rkt")
+
+(provide make-responsive clean-resized-images magick-available?)
+
+;; Depend on ImageMagick
+(define identify (find-executable-path "identify"))
+(define convert (find-executable-path "convert"))
+(define magick-available? (and identify convert))
+
+(module+ test
+  (require rackunit))
+
+(define (image-width path)
+  (string->number
+   (with-output-to-string
+     (λ () (system* identify "-format" "%w" path)))))
+
+(module+ test
+  (when magick-available?
+    (check-eq? (image-width (string->path "../example/img/800px-image.gif")) 800)))
+
+(define/contract (resize-image in new-width out)
+  (path? number? path? . -> . boolean?)
+  (prn1 "Resizing image ~a to ~a pixels... " (path->string in) new-width)
+  (define (resize in w out)
+    (system* convert in "-resize" (string->bytes/utf-8 (number->string w)) out))
+  (if (> (current-verbosity) 1)
+      (time (resize in new-width out))
+      (resize in new-width out)))
+
+(module+ test
+  (when magick-available?
+    (define out (make-temporary-file))
+    (test-eq? "resize" (begin
+                         (resize-image (string->path "../example/img/480px-image.gif") 10 out)
+                         (image-width out))
+              10)
+    (delete-file out))) ; catch exn?
+
+(define/contract (get-images image-path)
+  (path? . -> . (values pair? (listof pair?)))
+  (define output-path (string->path (current-image-output-dir)))
```

Comment:
This should change to something like `(build-path* (www/img-path) (current-image-output-dir))`.